code/datums/components/drift.dm

```diff
@@ -192,3 +216,52 @@
 		return
 	if(world.time < block_inputs_until)
 		return COMSIG_MOB_CLIENT_BLOCK_PRE_MOVE
+
+/datum/component/drift/proc/attempt_halt(mob/source, movement_dir, continuous_move, atom/backup)
+	SIGNAL_HANDLER
+
+	if (get_dir(source, backup) == movement_dir || source.loc == backup.loc)
+		if (drift_force >= INERTIA_FORCE_THROW_FLOOR)
+			source.throw_at(backup, 1, round(1 + (drift_force - INERTIA_FORCE_THROW_FLOOR) / INERTIA_FORCE_PER_THROW_FORCE), spin = FALSE)
+		qdel(src)
+		return
+
+	if (drift_force < INERTIA_FORCE_SPACEMOVE_GRAB)
+		qdel(src)
+		return
+
+	if (drift_force <= INERTIA_FORCE_SPACEMOVE_REDUCTION / source.inertia_force_weight)
+		glide_to_halt(round(20 / get_loop_delay(source)))
+		return COMPONENT_PREVENT_SPACEMOVE_HALT
+
+	drift_force -= INERTIA_FORCE_SPACEMOVE_REDUCTION / source.inertia_force_weight
+	drifting_loop.set_speed(get_loop_delay(source))
+	return COMPONENT_PREVENT_SPACEMOVE_HALT
+
+/datum/component/drift/proc/get_loop_delay(atom/movable/movable)
+	return (DEFAULT_INERTIA_SPEED / ((1 - INERTIA_SPEED_COEF) + drift_force * INERTIA_SPEED_COEF)) * movable.inertia_move_multiplier
```

Comment:
what is going on here? graph?

Reply:
![image](https://github.com/user-attachments/assets/bbe47408-c15f-4edd-9698-86d74952a162)
Basically a hyperbolic graph for delay, here's graph of actual speed (1/this) in tiles per tick
![image](https://github.com/user-attachments/assets/3e642312-1b15-4fac-99a1-a40b1657f12f)

Reply:
you should save this and put it in a comment somewhere relevant

Reply:
also yes this did help thank you